### DISCUSSION TITLE
Why inject is needed in this situation?

### DISCUSSION THREAD

**Discussion by bakura10** [inversify/InversifyJS#1521] [General]
Hi :),

Sorry if this is a stupid question, but I am not understanding why I need to use an @inject here. I am registering a MikroORM instance:

```
const orm = await MikroORM.init(config);

let container = new Container();
container.bind<MikroORM>(MikroORM).toConstantValue(orm);

container.resolve(Foo);
```

Where Foo is:

```
import {MikroORM} from "@mikro-orm/core";
import {inject, injectable} from "inversify";

@injectable()
export class Foo {
    constructor(manager: MikroORM) {
        console.log(manager);
    }
}
```

If I add @inject(MikroORM) this works, but I fail to understand why. Isn't the container supposed to automatically resolve this as this is a class name ?

I'm coming from PHP background, so I still need to adjust my way to Typescript, but I spent a few hours failing to understand why it is not working here.

Thanks!

**Comment by Jameskmonger** [comment]
Hey @bakura10, TypeScript is "transpiled" to JavaScript, which is untyped, therefore it lacks the type information at runtime.

The transpiled class from your example will look something like:

```javascript
class Foo {
    constructor(manager) {
        console.log(manager);
    }
}
```

Therefore the `@inject` is needed to attach the type information to the class at runtime.
